### PR TITLE
Use kapt

### PR DESCRIPTION
**Pull request by lvla**
## Overview (Required)
- use kapt, kotlin codes can reference classes generated by DataBinding, Orma, KVS-Schema

## why not use `kotlin-kapt`
[kotlin-kapt](https://blog.jetbrains.com/kotlin/2016/09/kotlin-1-0-4-is-here/) is unstable yet.

### CODE REVIEW

File: app/licenses.yml [hunk 1]
```diff
@@ -132,3 +132,11 @@
   license: The Apache Software License, Version 2.0
   licenseUrl: http://www.apache.org/licenses/LICENSE-2.0.txt
   url: https://github.com/chrisjenx/Calligraphy
+- artifact: org.jetbrains.kotlin:kotlin-runtime:+
```
Comment (gfx) [inline app/licenses.yml]: Kotlin runtime libraries can be merged into a single section like this:

```yaml
- artifact: org.jetbrains.kotlin:+:+
  name: Kotlin Runtime and Standard Library
  copyrightHolder: JetBrains Inc.
  license: The Apache Software License, Version 2.0 
```

File: app/licenses.yml [hunk 2]
```diff
@@ -132,3 +132,11 @@
...
+  name: kotlin-runtime
+  copyrightHolder: JetBrains Inc.
+  license: The Apache Software License, Version 2.0
+- artifact: org.jetbrains.kotlin:kotlin-stdlib:+
+  name: kotlin-stdlib
+  copyrightHolder: JetBrains Inc.
+  license: The Apache Software License, Version 2.0
```
Comment (gfx) [inline app/licenses.yml]: You should check the "ensure line feed at file end on Save" in Android Studio 😎 

screen shot 2017-02-07 at 9 17 44

File: app/licenses.yml (hunk 1 above)
Comment (lvla) [inline app/licenses.yml]: thanks, merged these😀

File: app/licenses.yml (hunk 2 above)
Comment (lvla) [inline app/licenses.yml]: it mean, i have to ensure line feed at file end ?

File: app/build.gradle [hunk 3]
```diff
@@ -236,10 +242,9 @@ dependencies {
     androidTestCompile 'com.android.support.test.espresso:espresso-core:2.2.1'
     androidTestCompile 'com.android.support.test.espresso:espresso-intents:2.2.1'
     testCompile 'junit:junit:4.12'
-    testCompile "org.jetbrains.kotlin:kotlin-stdlib:1.0.6"
     testCompile "org.jetbrains.kotlin:kotlin-test-junit:1.0.6"
-    testCompile 'com.taroid.knit:knit:0.1.2'
-    testCompile 'com.github.sys1yagi:kmockito:0.1.2'
+    testCompile('com.taroid.knit:knit:0.1.2') { exclude module:"kotlin-stdlib" }
+    testCompile('com.github.sys1yagi:kmockito:0.1.2') { exclude module:"kotlin-stdlib" }
```
Comment (gfx) [inline app/build.gradle]: Use resolution strategy, just like as `support-annotations` defined above the `dependencies`.

File: app/build.gradle [hunk 4]
```diff
@@ -165,12 +165,18 @@ android {
 configurations {
     all {
         resolutionStrategy {
+            force "org.jetbrains.kotlin:kotlin-stdlib:${kotlin_version}"
             force "com.android.support:support-annotations:${support_lib_version}"
         }
     }
 }
+kapt {
```
Comment (hotchemi) [inline app/build.gradle]: if i remember correctly this generate stub style is a bit old isn't it?
ref: https://blog.jetbrains.com/kotlin/2016/12/kotlin-1-0-6-is-here/

Comment (lvla) [inline app/build.gradle]: that's right, but i think `kotlin-kapt` plugin is unstable yet 😢

Comment (hotchemi) [inline app/build.gradle]: so you use this style for stability, gotcha.

**Comment by lvla** [comment]
exceeded the memory limit 😢

**Comment by gfx** [comment]
OMG.

Splitting tasks like this https://github.com/gfx/Android-Orma/blob/master/circle.yml#L33-L38 might make CI stable.

**Comment by lvla** [comment]
oh
there is a solution!?
thank you😲

**Comment by lvla** [comment]
like this?
```
test:
  override:
    - TERM=dumb ./gradlew clean assembleProductionDebug check jacocoTestReportDevelop
    - TERM=dumb find . -name app*debug.apk -exec mv {} $CIRCLE_ARTIFACTS/ \;
```

**Comment by gfx** [comment]
I'm afraid, like this:

```yaml
test:
  override:
    - TERM=dumb ./gradlew assembleProductionDebug
    - TERM=dumb ./gradlew check
    - TERM=dumb ./gradlew jacocoTestReportDevelop
```

**Comment by lvla** [comment]
you mean
```yml
test:
  override:
    - TERM=dumb ./gradlew assembleProductionDebug
    - TERM=dumb ./gradlew check
    - TERM=dumb ./gradlew jacocoTestReportDevelop
    - find . -name app*debug.apk -exec mv {} $CIRCLE_ARTIFACTS/ \;
```
k?

**Comment by gfx** [comment]
Yep! 👍

**Comment by lvla** [comment]
hmm...
a test using Robolectric failed
```
io.github.droidkaigi.confsched2017.ExampleUnitTest > addition_isCorrect FAILED
    org.apache.tools.ant.BuildException
        Caused by: org.apache.maven.artifact.resolver.MultipleArtifactsNotFoundException
```

looking for how to fix this, but i don't have no idea.
anyone has an idea??

**Comment by jmatsu** [comment]
@lvla 
Maybe CircleCI's network issue...
For now, triggered rebuilding.

File: circle.yml [hunk 5]
```diff
@@ -20,8 +20,10 @@ dependencies:
 
 test:
   override:
-    - ./gradlew clean assembleProductionDebug check jacocoTestReportDevelop
-    - find . -name app*debug.apk -exec cp {} $CIRCLE_ARTIFACTS/ \;
+    - TERM=dumb ./gradlew assembleProductionDebug
+    - TERM=dumb ./gradlew check
+    - TERM=dumb ./gradlew jacocoTestReportDevelop
+    - find . -name app*debug.apk -exec mv {} $CIRCLE_ARTIFACTS/ \;
```
Comment (jmatsu) [inline circle.yml]: @lvla 
Can you revert this line's change?  `cp` is correct command. 😄

**Comment by konifar** [comment]
Thank you for your contribution! Sorry to be late comment 🙇
Actually, I'm a little negative to introduce kapt. Because kapt is not stable as you know. 
After we introduce kapt, we might take time to solve its own problem.

Is there enough benefits to introduce it? I'm not sure, so I want to ask you

**Comment by lvla** [comment]
> After we introduce kapt, we might take time to solve its own problem.
agree

i thought kapt is not suitable to develop with many people
close this PR :D

**Comment by konifar** [comment]
Sorry :bowing_man:
Anyway, thank you for contribution!!

**Pull request closed without merging**